LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIStrategyMapItem_Region_LW.uc

```diff
@@ -7,37 +7,68 @@
 
 class UIStrategyMapItem_Region_LW extends UIStrategyMapItem_Region;
 
-//var StaticMesh RegionMesh;
-//const NUM_TILES = 3;
-//var StaticMeshComponent RegionComponents[NUM_TILES];
-//
-//var public localized String m_strLockedTT;
-//var public localized String m_strUnlockedTT;
-//var public localized String m_strContactTT;
-//var public localized String m_strOutpostTT;
-//var public localized String m_strControlTT;
-//
-//var public localized String m_strScanForIntelLabel;
-//var public localized String m_strScanForOutpostLabel; 
-//var public localized String m_strButtonMakeContact;
-//
-//var UIButton ContactButton; 
-//var UIButton OutpostButton;
-//var UIPanel BGPanel;
-//var UIScanButton ScanButton;
-//
-//var array<float> CumulativeTriangleArea;
-
 var localized string m_strOutpostTitle;
 var localized string m_strAlertLevel;
 var localized string m_strLiberatedRegion;
 var localized string m_strStaffingPinText;
 var localized string m_strStaffingPinTextMore;
 var localized string m_strMonthlyRegionalIncome;
 
+var bool CanMakeContact, CanScanContact, CanMakeRadioRelay, CanScanRadioRelay, RadioRelayInstalled;
+
 var string CachedRegionLabel;
 var string CachedHavenLabel;
 
+// KDM : Helper function which fills in variables that relate to a regions contact/radio relay status.
+// This has been made a function as it will likely be used in several places in the near future.
+simulated function UpdateRegionStatusVariables()
+{
+	CanMakeContact = class'UIUtilities_Strategy'.static.GetXComHQ().IsContactResearched() && (GetRegion().ResistanceLevel == eResLevel_Unlocked);
+	CanScanContact = GetRegion().bCanScanForContact;
+	CanMakeRadioRelay = class'UIUtilities_Strategy'.static.GetXComHQ().IsOutpostResearched() && (GetRegion().ResistanceLevel == eResLevel_Contact);
+	CanScanRadioRelay = GetRegion().bCanScanForOutpost;
+	RadioRelayInstalled = (GetRegion().ResistanceLevel == eResLevel_Outpost);
```

Comment:
Why are these object variables rather than individual functions, like `CanMakeContact()`? I can't imagine that caching the information helps performance much, and now we have to worry about ensuring `UpdateRegionStatusVariables()` is called at the right times, otherwise the cached data could become out of sync with the real source of truth.

Reply:
Seems like a reasonable idea.

Will change them to functions.